### PR TITLE
Correct handling of labels for regression models in attribute attacks

### PR DESCRIPTION
**Pull request by abigailgold**
# Description
Handling of true labels in some of the attribute inference attacks was not completely adapted to work correctly with regression labels. This was fixed so it works well for regression models as well.

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [X ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

New tests for regression models were added

**Test Configuration**:
- OS: 11.6.4
- Python version: 3.8
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [ X] My code follows the style guidelines of this project
- [ X] I have performed a self-review of my own code
- [ X] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [ X] My changes generate no new warnings
- [ X] I have added tests that prove my fix is effective or that my feature works
- [ X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: art/attacks/inference/attribute_inference/true_label_baseline.py [hunk 1]
```diff
@@ -59,6 +59,7 @@ def __init__(
         attack_model_type: str = "nn",
         attack_model: Optional["CLASSIFIER_TYPE"] = None,
         attack_feature: Union[int, slice] = 0,
+        is_regression: Optional[bool] = False,
```
Comment (beat-buesser) [inline art/attacks/inference/attribute_inference/true_label_baseline.py:62]: Would it be possible to derive this information from the type of `attack_model`?

**Review by beat-buesser — CHANGES_REQUESTED** [review]
Hi @abigailgold Thank you very much for these improvements. I think they are good but I have a few questions about the implementation.

File: art/attacks/inference/attribute_inference/true_label_baseline.py (hunk 1 above)
Comment (beat-buesser) [reply art/attacks/inference/attribute_inference/true_label_baseline.py:62]: Does the type of `attack_model` have to be updated to include `REGRESSOR_TYPE`?

**Review by beat-buesser — APPROVED** [review]
